File: src/tools/pmd-cpd/eval-repos/synthetic/javascript/cross_file_a.js
/**
 * JavaScript file A for cross-file duplication detection.
 */

function calculateOrderTotal(items) {
    let total = 0.0;
    for (const item of items) {
        const price = item.price || 0.0;
        const quantity = item.quantity || 1;
        const discount = item.discount || 0.0;
        const itemTotal = price * quantity * (1 - discount / 100);
        total += itemTotal;
    }
    return Math.round(total * 100) / 100;
}

function applyShippingCost(subtotal, country) {
    const shippingRates = {
        US: 5.99,
        CA: 8.99,
        UK: 12.99,
        DE: 14.99,
        FR: 14.99,
        AU: 19.99,
    };
    const baseRate = shippingRates[country] || 24.99;
    if (subtotal > 100) return subtotal;
    return subtotal + baseRate;
}

function applyTax(subtotal, state) {
    const taxRates = {
        CA: 0.0725,
        NY: 0.08,
        TX: 0.0625,
        FL: 0.06,
        WA: 0.065,
    };
    const rate = taxRates[state] || 0.0;
    const tax = subtotal * rate;
    return Math.round((subtotal + tax) * 100) / 100;
}

function formatOrderSummary(order) {
    const lines = [];
    lines.push("=".repeat(50));
    lines.push("ORDER SUMMARY");
    lines.push("=".repeat(50));
    lines.push(`Order ID: ${order.id || "N/A"}`);
    lines.push(`Customer: ${order.customerName || "Unknown"}`);
    lines.push(`Date: ${order.date || "Unknown"}`);
    lines.push("-".repeat(50));
    for (const item of order.items || []) {
        lines.push(`  ${item.name}: $${(item.price || 0).toFixed(2)}`);
    }
    lines.push("-".repeat(50));
    lines.push(`Subtotal: $${(order.subtotal || 0).toFixed(2)}`);
    lines.push(`Shipping: $${(order.shipping || 0).toFixed(2)}`);
    lines.push(`Tax: $${(order.tax || 0).toFixed(2)}`);
    lines.push(`Total: $${(order.total || 0).toFixed(2)}`);
    lines.push("=".repeat(50));
    return lines.join("\n");
}

module.exports = { calculateOrderTotal, applyShippingCost, applyTax, formatOrderSummary };

File: src/tools/pmd-cpd/eval-repos/synthetic/java/CrossFileA.java
/**
 * Java file A for cross-file duplication detection.
 */
package synthetic;

import java.util.*;

public class CrossFileA {

    public static double calculateOrderTotal(List<Map<String, Object>> items) {
        double total = 0.0;
        for (Map<String, Object> item : items) {
            double price = ((Number) item.getOrDefault("price", 0.0)).doubleValue();
            int quantity = ((Number) item.getOrDefault("quantity", 1)).intValue();
            double discount = ((Number) item.getOrDefault("discount", 0.0)).doubleValue();
            double itemTotal = price * quantity * (1 - discount / 100);
            total += itemTotal;
        }
        return Math.round(total * 100.0) / 100.0;
    }

    private static final Map<String, Double> SHIPPING_RATES = Map.of(
        "US", 5.99,
        "CA", 8.99,
        "UK", 12.99,
        "DE", 14.99,
        "FR", 14.99,
        "AU", 19.99
    );

    public static double applyShippingCost(double subtotal, String country) {
        double baseRate = SHIPPING_RATES.getOrDefault(country, 24.99);
        if (subtotal > 100) return subtotal;
        return subtotal + baseRate;
    }

    private static final Map<String, Double> TAX_RATES = Map.of(
        "CA", 0.0725,
        "NY", 0.08,
        "TX", 0.0625,
        "FL", 0.06,
        "WA", 0.065
    );

    public static double applyTax(double subtotal, String state) {
        double rate = TAX_RATES.getOrDefault(state, 0.0);
        double tax = subtotal * rate;
        return Math.round((subtotal + tax) * 100.0) / 100.0;
    }

    public static String formatOrderSummary(Map<String, Object> order) {
        List<String> lines = new ArrayList<>();
        lines.add("=".repeat(50));
        lines.add("ORDER SUMMARY");
        lines.add("=".repeat(50));
        lines.add("Order ID: " + order.getOrDefault("id", "N/A"));
        lines.add("Customer: " + order.getOrDefault("customerName", "Unknown"));
        lines.add("Date: " + order.getOrDefault("date", "Unknown"));
        lines.add("-".repeat(50));
        @SuppressWarnings("unchecked")
        List<Map<String, Object>> items = (List<Map<String, Object>>) order.getOrDefault("items", Collections.emptyList());
        for (Map<String, Object> item : items) {
            double price = ((Number) item.getOrDefault("price", 0.0)).doubleValue();
            lines.add(String.format("  %s: $%.2f", item.get("name"), price));
        }
        lines.add("-".repeat(50));
        lines.add(String.format("Subtotal: $%.2f", ((Number) order.getOrDefault("subtotal", 0.0)).doubleValue()));
        lines.add(String.format("Shipping: $%.2f", ((Number) order.getOrDefault("shipping", 0.0)).doubleValue()));
        lines.add(String.format("Tax: $%.2f", ((Number) order.getOrDefault("tax", 0.0)).doubleValue()));
        lines.add(String.format("Total: $%.2f", ((Number) order.getOrDefault("total", 0.0)).doubleValue()));
        lines.add("=".repeat(50));
        return String.join("\n", lines);
    }
}
